prow/gerrit/adapter/adapter.go

```diff
@@ -692,6 +697,28 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 	return nil
 }
 
+// lastRevisionWithCodeChange tries to find a revision with code change since last update.
+// There is no point repeating tests if no code change happened since last update.
+func lastRevisionWithCodeChange(change client.ChangeInfo, lastUpdate time.Time) (client.RevisionInfo, error) {
+	var revisionsWithCodeChange []client.RevisionInfo
+	for _, revision := range change.Revisions {
+		if revision.Kind != gerrit.NoCodeChange {
+			revisionsWithCodeChange = append(revisionsWithCodeChange, revision)
+		}
+	}
+	sort.Slice(revisionsWithCodeChange, func(i, j int) bool {
+		return revisionsWithCodeChange[i].Created.Time.After(revisionsWithCodeChange[j].Created.Time)
+	})
+
+	for _, revision := range revisionsWithCodeChange {
+		if revision.Created.Time.After(lastUpdate) {
```

Comment:
This only returns the first revision that was created after the last update. So if there are 2 revisions (both with code changes) we only return the first one. Can you add a comment why this is necessary here?

Also, if my analysis above is correct then the name of this function is misleading because we're not really getting the last revision at all (but the first one we find whose timestamp is after `lastUpdate`).

Reply:
Since we sort the revisions by creation time reversely, this actually returns the latest revision if we have 2 revisions with code change after last date.

Reply:
Thanks for the explanation. Nit: it would be good to capture this thought in a comment to prevent confusion for future readers of this code.